src/com/simulator/model/HeatingModule.java

```diff
@@ -51,21 +62,96 @@ public void displayTemperatureForRoom(String roomName, Profile currentUser){
+            currentRoom.getTemperature().setTemperatureOverridden(temperature);
+            currentRoom.setOverridden(true);
+            Logger.getInstance().outputToConsole(String.format("Temperature in %s is set to be at: %f currentTemperature: %f", roomName, temperature, currentRoom.getTemperature().getCurrentTemperature()));
             return true;
         }  
     }
+
+    @Override
+    public void updateLocation(Profile profile) {
+        return;
+    }
+
+    public void updateRoomTargetTemperature(Room room, double currentTempRoom, int currentHour){
+        if(room.getOverridden()){
+            room.getTemperature().setTemperatureTarget(room.getTemperature().getTemperatureOverridden());
+        }
+        else{
+            if(currentHour >=0 && currentHour < 8){
+                room.getTemperature().setTemperatureTarget(room.getTemperature().getTemperatureMorning());
+            }
+            else if(currentHour >= 8 && currentHour < 16){
+                room.getTemperature().setTemperatureTarget(room.getTemperature().getTemperatureDay());
+            }
+            else if(currentHour >= 16 && currentHour < 24){
+                room.getTemperature().setTemperatureTarget(room.getTemperature().getTemperatureNight());
+            }
+        }
+    }
+
+    public void checkToggleStatusHVAC(Room room, double currentTempRoom){
+        double targetTemperature = room.getTemperature().getTemperatureTarget();
+
+        if (currentTempRoom > targetTemperature + 0.25 || currentTempRoom < targetTemperature - 0.25){
+            room.setCurrentStateHVAC(true);
+        }
+        // Account for posibility of target.#5
+        else if(currentTempRoom >= targetTemperature - 0.05 && currentTempRoom <= targetTemperature + 0.05){
+            room.setCurrentStateHVAC(false);
+        }
+    }
+
+    public void checkSummerCooling(Room room, int currentMonth, double outdoorTemperature, double currentTempRoom){
+        // TODO check if in away mode before opening windows
+        if((currentMonth <= 9 && currentMonth >= 6) && currentTempRoom > outdoorTemperature + 0.25){
+            if(room.getCurrentStateHVAC()){
+                Logger.getInstance().outputToConsole("Disabling Air Conditioning: It is summer and cooler outdoors. Open all windows");
+            }
+            room.setCurrentStateHVAC(false);
+            room.openAllWindows();
+        }
+    }
+
+    public int getCurrentHour(Date date){
+        Calendar calendar = GregorianCalendar.getInstance();
+        calendar.setTime(date);
+
+        return calendar.get(Calendar.HOUR_OF_DAY);
+    }
+
+
+    @Override
+    public void updateTime(int time) {
+        SimulationParameters simulationInstance = SimulationParameters.getInstance();
+        int currentMonth = simulationInstance.getDate().getMonth();
+        int currentHour = simulationInstance.getTime() / 60;
```

Comment:
Under the new time control, dividing the time by 60 will give minutes rather than hours. / 3600 will give the hour